app/routes/v2/standards/index.js

```diff
@@ -14,7 +14,7 @@ module.exports = [{
       const response = await getPollingResponse(request.yar.id, '/standards')
       if (response) {
         console.info('Standards request received', response)
-        await cache.update('apply-journey', request.yar.id, { agreementNumber: response.agreementNumber, standards: response.standards })
+        await cache.update('apply-journey', request.yar.id, { agreementNumber: response.agreementNumber ?? `AG${new Date().getTime()}`, standards: response.standards })
```

Comment:
There is a generateAgreementNumber function in the agreement-number module

Reply:
agree